src/Xrefcheck/Verify.hs

```diff
@@ -155,14 +156,15 @@ verifyRepo
   let toScan = do
         (file, fileInfo) <- M.toList repoInfo
         guard . not $ any ((`isPrefixOf` file) . (root </>)) vcNotScanned
-        ref <- _fiReferences fileInfo
+        ref <- L.nubBy (\r1 r2 -> rLink r1 == rLink r2) $ _fiReferences fileInfo
```

Comment:
This means that if we have 2 references to the same link, and that link is broken, `xrefcheck` will only report 1 broken reference.

E.g.:

```md
[a](https://non-existent.reference/)
[b](https://non-existent.reference/)
```

```
=== Invalid references found ===                                                                                                           

  ➥  In file tests/markdowns/without-annotations/test.md
     bad reference (external) at src:7:1-36:
       - text: "a"
       - link: https://non-existent.reference/
       - anchor: -
     
     ⛂  ConnectionFailure Network.Socket.getAddrInfo (called with preferred socket type/protocol: AddrInfo {addrFlags = [AI_ADDRCONFIG], addrFamily = AF_UNSPEC, addrSocketType = Stream, addrProtocol = 0, addrAddress = <assumed to be undefined>, addrCanonName = <assumed to be undefined>}, host name: Just "non-existent.reference", service name: Just "443"): does not exist (Name or service not known)
```

We should report both references though (I've edited the issue's description to make this more clear).